pkg/services/govmomi/clustermodules/provider.go

```diff
@@ -87,19 +88,18 @@ func (cm *provider) DoesModuleExist(ctx context.Context, moduleID string, cluste
 		return false, nil
 	}
 
-	modules, err := cm.manager.ListModules(ctx)
-	if err != nil {
-		return false, err
+	_, err := cm.manager.ListModuleMembers(ctx, moduleID)
+	if err == nil {
+		return true, nil
 	}
 
-	for _, mod := range modules {
-		if mod.Cluster == clusterRef.Value && mod.Module == moduleID {
-			return true, nil
-		}
+	// Is there a better way of unwrap vAPI errors?
```

Comment:
```suggestion
```

govmomi in this case returns the private type `statusError`: https://github.com/vmware/govmomi/blob/v0.33.1/vapi/rest/client.go#L150

Reply:
I would suggest to ask govmomi folks if they are open to expose a util func to extract more details (like the status code)

Reply:
https://github.com/vmware/govmomi/pull/3310 :)